hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/snapshot/OMSnapshotCreateRequest.java

```diff
@@ -215,5 +230,21 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
     }
     return omClientResponse;
   }
-  
+
+  /**
+   * Removes the snapshot from the SnapshotChainManager.
+   */
+  private void removeSnapshotInfoFromSnapshotChainManager(
+      SnapshotChainManager snapshotChainManager,
+      SnapshotInfo info
+  ) {
+    try {
+      snapshotChainManager.deleteSnapshot(info);
+    } catch (IOException exception) {
+      LOG.warn("Failed to remove snapshot: {} from SnapshotChainManager.",
+          info, exception);
+      throw new IllegalStateException("Snapshot chain manager is corrupted.",
```

Comment:
We should not throw here. This method cleans up garbage when an exception creates a snapshot. If the error is thrown here, the original exception is hidden.

Reply:
Initially, I only logged the exception that but then after discussing with snapshot team, we decided to throw this because snapshot chain is corrupted and should be reset. The other reason is that, ideally `      snapshotChainManager.deleteSnapshot(info);` should not throw exception because it throws exception in case order of snapshot chair is messed up which should not happen needs investigation.

Reply:
Discussed offline and it is better to not throw exception because we lose the actual error in snapshot creation.